src/Features/CSharp/Portable/Snippets/CSharpProprSnippetProvider.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis.CSharp.Extensions.ContextQuery;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Snippets;
+using Microsoft.CodeAnalysis.Snippets.SnippetProviders;
+using Roslyn.Utilities;
+using static Microsoft.CodeAnalysis.CSharp.CSharpSyntaxTokens;
+
+namespace Microsoft.CodeAnalysis.CSharp.Snippets;
+
+[ExportSnippetProvider(nameof(ISnippetProvider), LanguageNames.CSharp), Shared]
+[method: ImportingConstructor]
+[method: Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+internal sealed class CSharpProprSnippetProvider() : AbstractCSharpAutoPropertySnippetProvider
+{
+    public override string Identifier => CSharpSnippetIdentifiers.RequiredProperty;
+
+    public override string Description => FeaturesResources.required_property;
+
+    protected override SyntaxToken[] GetAdditionalPropertyModifiers(CSharpSyntaxContext? syntaxContext) => [RequiredKeyword];
+
+    protected override bool IsValidSnippetLocationCore(SnippetContext context, CancellationToken cancellationToken)
+    {
+        if (!base.IsValidSnippetLocationCore(context, cancellationToken))
+            return false;
+
+        var syntaxContext = (CSharpSyntaxContext)context.SyntaxContext;
+        var precedingModifiers = syntaxContext.PrecedingModifiers;
+
+        // The required modifier can't be applied to members of an interface
+        if (syntaxContext.ContainingTypeDeclaration is InterfaceDeclarationSyntax)
+            return false;
+
+        if (syntaxContext.PrecedingModifiers.IsEmpty())
+            return true;
```

Comment:
If I see it correctly this `if` is also not needed. All subsequent checks won't trigger, so we will get to the end and return true there

Reply:
Yeah, you're right. Removed.